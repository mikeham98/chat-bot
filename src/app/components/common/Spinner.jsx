import React from 'react';
import styles from '../../../themes/components/spinner.scss';

const Spinner = () => <div className={styles.spinner}/>;

export default React.memo(Spinner);