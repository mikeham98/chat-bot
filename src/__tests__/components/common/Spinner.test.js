import React from 'react';
import {shallow} from 'enzyme';
import Spinner from '../../../app/components/common/Spinner'

describe('Spinner', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<Spinner {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {};
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a div with a className prop set to spinner', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
});