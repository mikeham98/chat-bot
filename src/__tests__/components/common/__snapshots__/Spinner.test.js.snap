// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner render should return a div with a className prop set to spinner 1`] = `
<div
  className="spinner"
/>
`;
